chk/modules/assertion/support.py

```diff
@@ -96,23 +104,105 @@ def case_AssertCount(self):
         """Asserts count of sequence on actual"""
         actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
         self.assertLessEqual(actual, self.expect)
 
+    def case_AssertListContains(self):
+        """Asserts expected exist in the actual."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == list, f"`{self.actual}` is not a list"
+        assert self.expect in actual, f"`{self.expect}` is not in the list"
+
+    def case_AssertMapHasKey(self):
+        """Asserts expected key exists in the map."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == dict, f"`{self.actual}` is not a map"
+        assert self.expect in actual, f"key `{self.expect}` is not in the map"
+
+    def case_AssertMapDoNotHasKey(self):
+        """Asserts expected key does not exist in the map."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == dict, f"`{self.actual}` is not a map"
+        assert self.expect not in actual, f"key `{self.expect}` is in the map"
+
+    def case_AssertStrContains(self):
+        """Asserts expected is in the actual string."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == str, f"`{self.actual}` is not a string"
+        assert self.expect in actual, f"key `{self.expect}` is not in the str"
+
+    def case_AssertMapKeyCount(self):
+        """Asserts expected is equal to the number of keys in the map."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == dict, f"`{self.actual}` is not a map"
+        assert self.expect == len(actual), f" the map has `{len(self.actual)}` keys"
+
+    def case_AssertMapHasKeys(self):
+        """Asserts expected is a subset of keys of the map."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == dict, f"`{self.actual}` is not a map"
+
+        has_keys, failed_key = True, None
+        for key in self.expect:
+            if key not in actual.keys():
+                failed_key, has_keys = key, False
+                break
+        assert has_keys, f"key `{failed_key}` is not present in the map"
+
+    def case_AssertMapDoNotHasKeys(self):
+        """Asserts expected is not a subset of map keys."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == dict, f"`{self.actual}` is not a map"
+
+        has_keys, failed_key = False, None
+        for key in self.expect:
+            if key in actual.keys():
+                failed_key, has_keys = key, True
+                break
+        assert not has_keys, f"key `{failed_key}` is present in the map"
```

Comment:
```suggestion
        assert not set(temp1) & set(temp2), f"at least one of the keys `{self.expect}` is present in the map"
```

Reply:
sure

Reply:
you can also, capture which key duplicates capturing the result, if you want

Reply:
Yes may be using set operation such as intersections